include/seqan3/range/container/dynamic_bitset.hpp

```diff
@@ -1580,62 +1587,44 @@ class dynamic_bitset
      */
     //!\brief Performs element-wise comparison.
     template <size_t cap>
-    //!\cond
-        requires cap <= bit_capacity /* resolves ambiguousness when comparing two dynamic_bitsets of unequal capacity */
-    //!\endcond
-    friend constexpr bool operator==(dynamic_bitset const & lhs, dynamic_bitset<cap> const & rhs) noexcept
+    constexpr bool operator==(dynamic_bitset<cap> const & rhs) const noexcept
```

Comment:
why this change? I thought we always declare comprison operators as friends?

Reply:
I try to remember, but there was a problem. Enrico wrote: resolves ambiguousness when comparing two dynamic_bitsets of unequal capacity. I think this is a different issue.

I think the problem was that

```c++
friend constexpr bool operator==(dynamic_bitset const & lhs, dynamic_bitset<cap> const & rhs) noexcept;
```

means: my current class instance `dynamic_bitset` befriends `operator==` where first argument is my current class instance and the second argument is any other class instance of dynamic_bitset. (NOTE: they don't befriend this function overload)

This means that `operator==` is only allowed to access `lhs`, but not `rhs`, because `rhs` is not necessary the same type as the befriending `lhs`.

TLDR; Only lhs type befriends that class, rhs type does not.

----

I'm not to sure if we should always use `friend` declarations. 

I read https://blogs.accu.org/?p=1974 and then tried to play around with https://godbolt.org/z/ZmIqxA.

AFAIU we befriend to have the `hidden friend` property that means that a friend declared function WILL not be found by ADL of other types. And it allows to implicitly converts the left hand side in your type (if you have an implicit conversion constructor).

That means we lose in this case the implicit conversion of the left hand side. If it's important we need to figure out something else.

If we want this, we need to declare
```c++
    template <size_t cap1, size_t cap2>
    friend inline constexpr bool operator==(dynamic_bitset<cap1> const & lhs, dynamic_bitset<cap2> const & rhs) noexcept;
```
and define the friend function outside once, but then any implicit conversation from any other type to bitset would use that comparison operator. 

Making the private members protected would work.

Reply:
Maybe the more killer argument:

```c++
#include <bitset>

int main()
{
    std::bitset<5> a{};
    a == 5; // works
    // 5 == a; // does not works (because operator== is a member)
    return 0;
}
```

Reply:
Phew I am not sure if I understood everything :see_no_evil: 
let's talk about this in person

Reply:
yeah, I talked with @rrahn and he was in favour for the member instead of friend. If I understood @eseiler correctly, than he doesn't mind. Maybe we should put it as a discussion for next monday.

Reply:
We know use the public api.